src/style/style.js

```diff
@@ -309,7 +309,7 @@ class Style extends Evented {
         }
 
         this._loaded = true;
-        this.stylesheet = json;
+        this.stylesheet = extend({}, json);
```

Comment:
Nice fix 👍

Reply:
Thanks for the review @SnailBones!!  all fixes to unit tests pushed! I chatted with @ansis today who recommended doing deep clone instead of shallow clone as nested objects would still be manipulated with `extend`